### PR TITLE
Move pallenelib to inside the Lua interpreter

### PR DESCRIPTION
**Pull request by hugomg**
This set of commits gets rid of the separate pallenelib library. This is part of the ongoing effort to simplify the build and installation process for Pallene (see #16, #153). The pallene_core files are now going to be part of our the custom installation of Lua.

Unfortunately, for some reason it is not working. When I compile the custom Lua, the
`pallene_*` symbols are not being exported by the custom Lua executable. We'll need to figure out what is the problem.

TODO:
- [x] Find out why the symbols are not being exported
- [x] Fix compilation warnings that appeared because of -Wextra
- [x] Update the patch file we use when updating to a new Lua version
- [x] Delete the "runtime" folder, and find a new place to store the patch file
- [x] Update the installation instructions in the README.

### CODE REVIEW

**Comment by hugomg** [comment]
I think this might be one of those PRs where it is easier to understand if it is read one commit at a time.

File: vm/src/lua.c [hunk 1]
```diff
@@ -657,3 +657,16 @@ int main (int argc, char **argv) {
...
 
+/*
+** PALLENE LINKER HACK
+**
+** When we build the final "lua" executable, bundling it with the "liblua.a" library, the linker
+** only includes symbols from the object files where at least one of the symbols of the module is
+** used by the executable itself. This is a problem for the Pallene runtime library, because those
+** symbols are only used by extension modules, which we expect to dynamically link. One hacky
+** workaround I found is to force the "lua" executable to reference one of the symbols from the
+** pallene_code, as done below.
```
Comment (srijan-paul) [inline vm/src/lua.c:668]: This was supposed to say `pallene_core` perhaps?

File: vm/src/pallene_core.c [hunk 2]
```diff
@@ -159,6 +161,7 @@ void pallene_grow_array(lua_State *L, const char* file, int line, Table *arr, un
 
 void pallene_io_write(lua_State *L, TString *str)
 {
+    (void) L;
```
Comment (srijan-paul) [inline vm/src/pallene_core.c:164]: I wonder what this does?

File: pallene/coder.lua [hunk 3]
```diff
@@ -1265,7 +1265,7 @@ gen_cmd["GetTable"] = function(self, cmd, _func)
 
     return util.render([[
         {
-            static size_t cache = UINT_MAX;
+            static int cache = UINT_MAX;
```
Comment (srijan-paul) [inline pallene/coder.lua:1268]: should this be `INT_MAX` now?

**Review by srijan-paul — CHANGES_REQUESTED** [review]
I'm wondering if it would be possible to keep the pallene files in `vm/pallene/` instead of `vm/src/`? and if that would be better?

File: README.md [hunk 4]
```diff
@@ -70,18 +70,18 @@ $ echo 'eval `luarocks path`' >> ~/.bashrc
...
-Pallene must be run against a custom-built version of the Lua interpreter, as
-well as the Pallene runtime library. Both of these are written in C and must be
-compiled before the Pallene compiler can be used.
+Pallene must be run against a custom-built version of the Lua interpreter.
+This custom version of Lua 5.4 doesn't have to be the same one that you will use to run the compiler itself,
+or to install the Luarocks packages.
 
-These two components can be built through the Makefile we provide. The command
-to be used depends on your operating system:
+To compile the custom version of Lua, follow the instructions found the vm/doc/readme.html.
```
Comment (srijan-paul) [inline README.md:79]: > instructions found the

"Instructions found in the [path]" perhaps?

**Review by srijan-paul — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (hugomg) [reply README.md:79]: Apparently, Github displays the raw HTML if we link to that. I think I'm going to link to the lua.org website instead.

**Comment by hugomg** [comment]
Because the Lua Makefile cds into the "src" directory do to it's thing, I think it's clearer if we put all our files there instead of on a separate file. But I'm glad you asked, because I found a separate bug --  I had forgotten to modify the parent Makefile to tell it to install the additional header files.

**Review by srijan-paul — APPROVED** [review]

**Pull request merged**